suite/tests/linux/syscall_pwait.c

```diff
@@ -50,19 +50,34 @@
 #include <time.h>
 #include <poll.h>
 
+#define MAX_SIG 32
+
+typedef struct {
+    sigset_t *sigmask;
+    size_t sizemask;
+} data_t;
+
+static struct timespec sleeptime;
+
 static void
 signal_handler(int sig, siginfo_t *siginfo, void *context)
 {
     print("signal received: %d\n", sig);
 }
 
+static bool
+compare_sigmasks(sigset_t *sig_a, sigset_t *sig_b)
+{
+    for (int i = 0; i < MAX_SIG; ++i) {
+        if (sigismember(sig_a, i) != sigismember(sig_b, i))
+            return false;
```

Comment:
A simpler way is how the existing tests are doing this, by zeroing the padding: run `git grep -A 2 check_mask`

Reply:
Sure! Done